samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -358,90 +359,98 @@ void Launcher::DoTranscoding()
+                           << m_pThreadContextArray[i]->pPipeline->GetSessionText()
+                           << MSDK_STRING("] failed with status ")
+                           << StatusToString(m_pThreadContextArray[i]->transcodingSts)
+                           << MSDK_STRING(" shutting down the application...")
+                           << std::endl << std::endl;
+                        msdk_printf(MSDK_STRING("%s"), ss.str().c_str());
+
+                        for (auto context : m_pThreadContextArray)
                         {
-                            msdk_stringstream ss;
-                            ss << MSDK_STRING("\n\n session ") << i << MSDK_STRING(" [") << m_pSessionArray[i]->pPipeline->GetSessionText() << MSDK_STRING("] failed with status ") << StatusToString(m_pSessionArray[i]->transcodingSts)
-                               << MSDK_STRING(" shutting down the application...") << std::endl << std::endl;
-                            msdk_printf(MSDK_STRING("%s"), ss.str().c_str());
-
-                            for (size_t j = 0; j < m_pSessionArray.size(); j++)
-                            {
-                                m_pSessionArray[j]->pPipeline->StopSession();
-                            }
+                            context->pPipeline->StopSession();
                         }
                     }
-
-                    // Now clear its handle and thread info
-                    MSDK_SAFE_DELETE(m_HDLArray[i]);
-                    m_HDLArray[i] = NULL;
+                }
+                else if (m_pThreadContextArray[i]->transcodingSts > MFX_ERR_NONE)
+                {
+                    msdk_stringstream ss;
+                    ss << MSDK_STRING("\n\n session ") << i << MSDK_STRING(" [")
+                    << m_pThreadContextArray[i]->pPipeline->GetSessionText()
+                    << MSDK_STRING("] returned warning status ")
+                    << StatusToString(m_pThreadContextArray[i]->transcodingSts)
+                    << std::endl << std::endl;
+                    msdk_printf(MSDK_STRING("%s"), ss.str().c_str());
+                }
+            }
+            else
+            {
+                aliveSessionsCount++;
+                if (m_pThreadContextArray[i]->pPipeline->IsOverlayUsed()) {
+                    aliveOverlaySessionsCount++;
                 }
             }
         }
 
         // If all sessions are already stopped (no matter with error or not) - we need to forcibly stop all overlay sessions
-        if (aliveSessionsCount <= nOverlayThreads + 1)
+        if (aliveOverlaySessionsCount && aliveSessionsCount <= aliveOverlaySessionsCount + 1)
         {
             // Sending stop message
-            for (size_t i = 0; i < totalSessions; i++)
+            for (auto context : m_pThreadContextArray)
             {
-                if (m_HDLArray[i] && m_pSessionArray[i]->pPipeline->IsOverlayUsed())
+                if (!context->handle.valid())
```

Comment:
Separate Stop() API for ThreadTranscodeContext?

Reply:
StopSession is already a Stop API for ThreadTranscodeContext and it looks pretty nice (see lines 401-404). This one is special case for overlay sessions. It's not clear for me if we need it in API